generator/getters/gen.go

```diff
@@ -0,0 +1,133 @@
+	into.AddHelp(DefaultFieldMarker,
+		markers.SimpleHelp("Devfile", "indicates the default value of a boolean pointer field"))
+	return genutils.RegisterUnionMarkers(into)
+
+}
+
+func (Generator) CheckFilter() loader.NodeFilter {
+	return func(node ast.Node) bool {
+		// ignore interfaces
+		_, isIface := node.(*ast.InterfaceType)
+		return !isIface
+	}
+}
+
+// getterInfo stores the info to generate the getter method
+type getterInfo struct {
+	funcName   string
+	defaultVal string
+}
+
+// Generate generates the artifacts
+func (g Generator) Generate(ctx *genall.GenerationContext) error {
+	for _, root := range ctx.Roots {
+		ctx.Checker.Check(root)
+		root.NeedTypesInfo()
+
+		typesToProcess := orderedmap.NewOrderedMap()
+		if err := markers.EachType(ctx.Collector, root, func(info *markers.TypeInfo) {
+			if info.Markers.Get(GetterTypeMarker.Name) != nil {
+				var getters []getterInfo
+				for _, field := range info.Fields {
+					defaultVal := field.Markers.Get(DefaultFieldMarker.Name)
+					if defaultVal != nil {
+						if _, err := strconv.ParseBool(defaultVal.(string)); err != nil {
+							root.AddError(fmt.Errorf("devfile:default:value marker specified on %s/%s does not have a true or false value.  Value is %s", info.Name, field.Name, defaultVal.(string)))
+						}
+
+						//look for boolean pointers
+						if ptr, isPtr := field.RawField.Type.(*ast.StarExpr); isPtr {
+							if ident, ok := ptr.X.(*ast.Ident); ok {
+								if ident.Name == "bool" {
+									getters = append(getters, getterInfo{
+										field.Name,
+										defaultVal.(string),
+									})
+								} else {
+									root.AddError(fmt.Errorf("devfile:default:value marker is specified on %s/%s which is not a boolean pointer", info.Name, field.Name))
+								}
+							}
+						} else {
+							root.AddError(fmt.Errorf("devfile:default:value marker is specified on %s/%s which is not a boolean pointer", info.Name, field.Name))
+						}
+
+					}
+				}
+				if len(getters) > 0 {
+					typesToProcess.Set(info, getters)
+				} else {
+					root.AddError(fmt.Errorf("type %s does not have the field marker, devfile:default:value specified on a boolean pointer field", info.Name))
+				}
+				return
+			}
+
+		}); err != nil {
+			root.AddError(err)
+			return nil
+		}
+
+		genutils.WriteFormattedSourceFile("getters", ctx, root, func(buf *bytes.Buffer) {
+			for elt := typesToProcess.Front(); elt != nil; elt = elt.Next() {
+				cmd := elt.Key.(*markers.TypeInfo)
+				fields := elt.Value.([]getterInfo)
+				for _, getter := range fields {
+					fName := getter.funcName
+					defaultVal := getter.defaultVal
+					getterMethod := `
+// Get` + fName + ` returns the value of the boolean property.  If unset, it's the default value specified in the devfile:default:value marker` + `
+func (in *` + cmd.Name + `) Get` + fName + `() bool  {
+	return getBoolOrDefault(in.` + fName + `,` + defaultVal + `)}`
+					buf.WriteString(getterMethod)
```

Comment:
We could consider using `fmt.Sprintf` here, but it's optional. 

```suggestion
					fmt.Sprintf(`
// Get %[1]s returns the value of the boolean property. If unset, it's the default value specified in the devfile:default:value marker
func (in *%[2]s) Get%[1]s() bool {
	return getBoolOrDefault(in.%[1]s, %[3]s)
}`, fName, cmd.Name, defaultVal)
					buf.WriteString(getterMethod)
```
(looks like markdown formatting is broken by including a backtick in the sample in the sample :/ )